prusti-encoder/src/encoders/mir_pure.rs

```diff
@@ -1135,7 +1144,13 @@ mod rev_doms {
         pub fn new<'a, 'vir>(blocks: &'a mir::BasicBlocks<'vir>) -> Self {
             let no_succ_blocks = blocks
                 .iter_enumerated()
-                .filter(|(_, data)| data.terminator().successors().next().is_none())
+                .filter(|(_, data)| {
+                    /* The below prevents e.g. a SwitchInt with a branch going to
+                       an Unreachable terminator from having a join point, so do
+                       not treat such terminators as return blocks. */
```

Comment:
```suggestion
                    // The below prevents e.g. a SwitchInt with a branch going to
                    // an Unreachable terminator from having a join point, so do
                    // not treat such terminators as return blocks.
```

Reply:
The below being the `matches!`? If so, the commented out `data.terminator()...` should be removed?

Reply:
No, it's the commented out `data.terminator()` - I'm not sure exactly what the correct thing to put there is yet, but I think any terminator which actually exits out of the function in a "controlled" way

Reply:
And I use this comment to distinguish it from the line below on purpose